### PR TITLE
PXC-4050: Merged upstream Galera tag 'release_26.4.13'

### PR DESCRIPTION
**Pull request by kamil-holubicki**

### CODE REVIEW

File: gcs/src/gcs_gcomm.cpp [hunk 1]
```diff
@@ -223,10 +222,6 @@ class GCommConn : public Toplay
         {
             mutex_.unlock();
         }
-||||||| 63116854
-=======
-        delete tp_;
```
Comment (dutow) [inline gcs/src/gcs_gcomm.cpp]: We don't need this delete? According to the upstream commit this fixes a memory leak. Edit: nevermind, we already fixed that leak.

Comment (kamil-holubicki) [reply gcs/src/gcs_gcomm.cpp]: yes, this delete is at the beginning of the method

File: gcache/src/gcache_params.cpp [hunk 2]
```diff
@@ -57,21 +57,15 @@ gcache::GCache::Params::register_params(gu::Config& cfg)
...
-    cfg.add(GCACHE_PARAMS_RECOVER,         GCACHE_DEFAULT_RECOVER);
+    cfg.add(GCACHE_PARAMS_RECOVER, GCACHE_DEFAULT_RECOVER,
+            gu::Config::Flag::read_only | gu::Config::Flag::type_bool);
 #ifdef PXC
     cfg.add(GCACHE_PARAMS_KEEP_PAGES_COUNT, GCACHE_DEFAULT_KEEP_PAGES_COUNT);
     cfg.add(GCACHE_PARAMS_FREEZE_PURGE_SEQNO, GCACHE_DEFAULT_FREEZE_PURGE_SEQNO);
 #endif /* PXC */
     cfg.add(GCACHE_PARAMS_ENCRYPTION, GCACHE_DEFAULT_ENCRYPTION);
     cfg.add(GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE, GCACHE_DEFAULT_ENCRYPTION_CACHE_PAGE_SIZE);
     cfg.add(GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE, GCACHE_DEFAULT_ENCRYPTION_CACHE_SIZE);
```
Comment (venkatesh-prasad-v) [inline gcache/src/gcache_params.cpp]: Please check

- `GCACHE_PARAMS_ENCRYPTION` , `GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE` and `GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE` must be inside `#ifdef PXC` section as these exist only in PXC only.
- Add `gu::Config::Flag::type_bool` flag to `GCACHE_PARAMS_ENCRYPTION`
- Add `gu::Config::Flag::type_integer` flag to `GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE` and `GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE` and set the read only flags if necessary.

File: gcomm/src/conf.cpp [hunk 3]
```diff
@@ -125,19 +129,11 @@ gcomm::Conf::register_params(gu::Config& cnf)
...
-#endif /* PXC */
-||||||| 63116854
-    GCOMM_CONF_ADD (COMMON_BASE_HOST_KEY);
-    GCOMM_CONF_ADD (COMMON_BASE_PORT_KEY);
-=======
     GCOMM_CONF_ADD (BaseHost);
     GCOMM_CONF_ADD (BasePort);
->>>>>>> codership/galera/4.x
+#ifdef PXC
+    GCOMM_CONF_ADD (CommonBaseDir);
```
Comment (venkatesh-prasad-v) [inline gcomm/src/conf.cpp]: Cane we rename it to just `BaseDir`, just like `BaseHost` and `BasePort`?

File: gcache/src/gcache_params.cpp (hunk 2 above)
Comment (venkatesh-prasad-v) [reply gcache/src/gcache_params.cpp]: Also please set the appropriate flags for `GCACHE_PARAMS_KEEP_PAGES_COUNT` and `GCACHE_PARAMS_FREEZE_PURGE_SEQNO` as well.

Comment (venkatesh-prasad-v) [reply gcache/src/gcache_params.cpp]: Also, the same needs to be done in Allocator params as well. See `gu::Allocator::register_params()` in `galerautils/src/gu_alloc.cpp`. Probably read only flag is suitable here.

Comment (kamil-holubicki) [reply gcache/src/gcache_params.cpp]: Done

File: gcomm/src/conf.cpp (hunk 3 above)
Comment (kamil-holubicki) [reply gcomm/src/conf.cpp]: Done

**Review by venkatesh-prasad-v — APPROVED** [review]

**Review by dutow — APPROVED** [review]

**Pull request merged**